CMakeCommon.cmake

```diff
@@ -221,6 +221,25 @@ function (GenerateAddOnProject target acVersion devKitDir addOnName addOnSources
     target_include_directories (${target} SYSTEM PUBLIC ${devKitDir}/Inc)
     target_include_directories (${target} PUBLIC ${addOnSourcesFolder})
 
+    # use GSRoot custom allocators consistently in the Add-On
+    get_filename_component(new_hpp "${devKitDir}/Modules/GSRoot/GSNew.hpp" REALPATH)
+    get_filename_component(malloc_hpp "${devKitDir}/Modules/GSRoot/GSMalloc.hpp" REALPATH)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        target_compile_options(
+            "${target}" PRIVATE
+            "SHELL:/FI \"${new_hpp}\""
```

Comment:
For anyone else wondering what these `SHELL:` prefixes do: https://cmake.org/cmake/help/latest/command/target_compile_options.html#option-de-duplication